pilot/pkg/model/conversion.go

```diff
@@ -87,28 +87,17 @@ func (ps *ProtoSchema) FromJSON(js string) (proto.Message, error) {
 	if err != nil {
 		return nil, err
 	}
-	if err = ApplyJSON(js, pb, true); err != nil {
+	if err = ApplyJSON(js, pb); err != nil {
 		return nil, err
 	}
 	return pb, nil
 }
 
-// ApplyJSON unmarshals a JSON string into a proto message. Unknown fields will produce an
-//// error unless strict is set to false.
-func ApplyJSON(js string, pb proto.Message, strict bool) error {
+// ApplyJSON unmarshals a JSON string into a proto message.
+func ApplyJSON(js string, pb proto.Message) error {
 	reader := strings.NewReader(js)
-	m := jsonpb.Unmarshaler{}
-	if err := m.Unmarshal(reader, pb); err != nil {
-		if strict {
-			return err
-		}
-
-		log.Warnf("Failed to decode proto: %q. Trying decode with AllowUnknownFields=true", err)
-		m.AllowUnknownFields = true
-		reader.Reset(js)
-		return m.Unmarshal(reader, pb)
-	}
-	return nil
+	m := jsonpb.Unmarshaler{AllowUnknownFields: true}
```

Comment:
Accepting unknown fields without warning is not good either.
We should parse this first in strict mode, log error and then parse it leniently.

Reply:
why? Then it will flood the logs.. unnecessarily. Especially since this gets called for every time we read the CRDs and config maps from k8s. Sometimes several times a second.

Reply:
because swallowing error without any logs is bad. A typo will be silently ignored.
If the file has not changed, we don’t need to log again and again. Are we keeping track of last error encounter while processing a resource?
@ozevren

Reply:
make sense, some info level log SG

Reply:
Debug log.. not info.. Remember this is called periodically, as well as for every update from k8s

Reply:
ok

Reply:
done